generate.js

```diff
@@ -450,73 +454,128 @@ let update_output = function () {
+    node.rows = num_new_lines;
   }
-});  // ends listen for input
+}
 
-let ignore_node = document.getElementById( 'ignore_anywhere' );
+
+// TODO: Rearrange the sections to be closer to their values/defaults
+// TODO: Refactor name to 'exclude' instead of 'ignore'
+// ============================
+// Excluding rows based on their contents
+// ============================
+// keys_to_ignore tests only the variable names, not the whole contents of the row.
+// THIS VALUE STAYS THE SAME ALWAYS. It's hard-coded in here.
+// Show this list to the user so they might better understand what's going on.
+let keys_to_exclude_node = document.getElementById( 'auto_excluded_keys_only' );
+keys_to_exclude_node.innerText = JSON.stringify( keys_to_ignore );
+
+// 
+// ignore_anywhere CAN change. Show current value and allow changes.
+let exclude_anywhere_node = document.getElementById( 'ignore_anywhere' );
 let ignore_anywhere_default_alphabetical = ignore_anywhere_default.sort(function (a, b) {
     if (a > b) { return 1; }
     if (b > a) { return -1; }
     return 0;
 });
+let ignore_anywhere = ignore_anywhere_default_alphabetical;
+exclude_anywhere_node.value = JSON.stringify( ignore_anywhere, null, 2 );
+let ignore_error = document.querySelector( 'section#extra_options_container .error_output' );
+
 
-let handle_ignore_error = function () {
-  if ( ignore_node.value ) {
+// Listen for changes that customize the ignore_anywhere list
+document.body.addEventListener( 'input', function( event ) {
+  fit_textarea_to_content( event.target );
+  if ( event.target.id === 'ignore_anywhere' ) {
+    update_exclude_anywhere( event.target.value );
+    update_output();
+  }
+});  // ends listen for input
+
+let update_custom_exclusion_error = function ( err_msg ) {
+  // Either make the error visible or hide it, depending on what's needed
+  // Needs a better name or something too
+  if ( err_msg !== '' ) { console.error( err_msg ); }
+  ignore_error.innerText = err_msg;
+}
+
+let update_exclude_anywhere = function ( new_value ) {
+  // Set the value for ignore_anywhere and size of textarea. Handle errors.
+  // Do not change contents of textarea or user will not be able to edit it.
+  if ( new_value ) {
     try {
-      ignore_anywhere = JSON.parse( ignore_node.value );
-      ignore_warning.classList.remove('error');
-      ignore_error.innerText = '';
-    } catch ( err) {
-      console.warn( err );
-      ignore_warning.classList.add('error');
-      ignore_error.innerText = err;
-      ignore_anywhere = ignore_anywhere_default_alphabetical;  // use default
+      let maybe_exclusion_list = new_value;
+      if ( typeof new_value === `string` ) { maybe_exclusion_list = JSON.parse( new_value ); }
+      if ( !Array.isArray( maybe_exclusion_list )) {
+        // Does not yet test that every item is a string
+        let error = `Error: The data is not a JSON list of strings. This tool can only use a list of strings to exclude rows in the story table.`
```

Comment:
I'm not sure that this is working as intended... This is how I tested it: 

1. Create JSON file with contents `{"bob"}` (this is an object, not an array, so should error)
2. Upload this JSON file in the box that says "Customize the list of text that will cause a row in the table to be excluded."
3. The error message I see is `Could not load "tests.feature". SyntaxError: Unexpected token } in JSON at position 6`, which is not as helpful as the text here

Reply:
Ah nvm, my object was poorly constructed, duh. This works if the file I upload says `{"bob": "duck"}`. Nothing to see here.